btree/btree.c

```diff
@@ -3310,34 +3310,45 @@ static void vkvv_make(struct slot *slot, struct m0_be_tx *tx)
 	 *        given index. It is assumed that the possibility whether this
 	 *        can occur or not is determined before calling this function.
 	 *
+	 *        We will maintain an extra record than vkvv_used in the
+	 *        directory. Maintaining this extra offset value will help in
+	 *        size calculations.
+	 *
 	 *        start_key_addr = vkvv_key(slot->s_node, index)
 	 *        start_val_addr = vkvv_val(slot->s_node, index)
 	 *        count = num of records i.e. vkvv_used
 	 *
-	 *        idx = index
-	 *        while(idx <= count) {
-	 *          total_key_size += vkvv_keysize(slot->s_node, idx)
-	 *          total_val_size += vkvv_valsize(slot->s_node, idx)
-	 *          idx++
-	 *        }
+	 *        if(index==count) {
+	 *          //No need to do memmove() as data will added to the end of
+	 *          //current series of keys and values. Just update the
+	 *          //directory to keep a record of the next possible offset.
+	 *          dir_rec[index+1].key_offset = dir_rec[index].key_offset + incoming_key_size
+	 *          dir_rec[index+1].val_offset = dir_rec[index].val_offset + incoming_val_size
```

Comment:
It should be dir_rec[index+1].val_offset = dir_rec[index].val_offset - incoming_val_size
Please check

Reply:
No Radha, It wont be.

See, this calculation is to maintain the absolute offset for all the values.
The offset for values are considered from the end.
Hence, we will be adding the size to the last value's offset.
When we will be calculating the address using the offset at a particular index, at that moment we will subtract the offset from the end address to get the actual address for the value in node space.